Realm/RLMSyncSession.h

```diff
@@ -56,6 +102,32 @@ NS_ASSUME_NONNULL_BEGIN
 /// session.
 - (nullable RLMSyncConfiguration *)configuration;
 
-NS_ASSUME_NONNULL_END
+/**
+ Register a progress notification block. Multiple blocks can be registered on the same session at once. All blocks
+ will be dispatched to the main queue.
+
+ The token returned by this method must be retained as long as progress notifications are desired, and the `-stop`
+ method should be called on it when notifications are no longer needed.
+
+ If no token is returned, the session was not in a state where it could accept progress notifiers, or the notifier was
+ not a streaming notifier, was called immediately, and will not be called again since there is no additional progress
```

Comment:
What is a "streaming notifier"? Why would the block be called immediately? Why would the session not be in a state where it can accept progress notifiers?